toolchain/check/testdata/interop/cpp/function/class.carbon

```diff
@@ -24,13 +24,25 @@ auto foo(C) -> void;
 
```

Comment:
This is a `fail_todo_` test. Is there more to do here? The diagnostic we're producing seems pretty reasonable.

Likewise for the `fail_todo_` tests below using this incomplete type -- can the `todo_` be removed?